src/mgmt/srv_system.c

```diff
@@ -631,7 +671,7 @@ ds_mgmt_join_handler(struct mgmt_join_in *in, struct mgmt_join_out *out)
 			goto out_lock;
 		}
 		D_DEBUG(DB_TRACE, "rank %u rejoined\n", rank);
-		notify_map_distributord(svc);
+		ds_rsvc_request_map_dist(&svc->ms_rsvc);
 		rc = 0;
 		goto out_lock;
```

Comment:
the ds_rsvc_request_map_dist() and out->jo_rank assignment are the same as some steps that are done just before the out_lock: label. Is it worth having this code jump to a new label just before out_lock and make the ds_rsvc_request_map_dist() call there, on one line of code in this function? Realize out->jo_flags is not the same here versus at that later part of the function.